### ISSUE TITLE
Extract metadata from firmware file

### ISSUE THREAD

**Issue by FloSchels** [amosproj/amos2022ss01-firmware-downloader#26]
## User story
1. As a user
2. I want the metadata extracted from the firmware after the download and a hash of the file to be calculated
3. So that it can be saved to the database and be compared if changes happen

## Acceptance criteria
* All relevant metadata is extracted (if given):
* FileName, CreationDate, LastEditDate, FileSize (in Bytes)
* Check if signature can be extracted as well
* Hash value (checksum) of file

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
